src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -221,13 +221,42 @@ private static MethodSymbol GetEntryPoint(CSharpCompilation compilation, PEModul
                 !hasDeclarationErrors &&
                 !diagnostics.HasAnyErrors())
             {
-                var body = synthesizedEntryPoint.CreateBody();
+                BoundStatement body = synthesizedEntryPoint.CreateBody();
+
+                var dynamicAnalysisSpans = ImmutableArray<SourceSpan>.Empty;
+                var diagsForCurrentMethod = DiagnosticBag.GetInstance();
+                VariableSlotAllocator lazyVariableSlotAllocator = null;
+                var lambdaDebugInfoBuilder = ArrayBuilder<LambdaDebugInfo>.GetInstance();
+                var closureDebugInfoBuilder = ArrayBuilder<ClosureDebugInfo>.GetInstance();
+                StateMachineTypeSymbol stateMachineTypeOpt = null;
                 const int methodOrdinal = -1;
+
+                var loweredBody = LowerBodyOrInitializer(
+                    synthesizedEntryPoint,
+                    methodOrdinal,
+                    body,
+                    null,
+                    new TypeCompilationState(synthesizedEntryPoint.ContainingType, compilation, moduleBeingBuilt),
+                    false ,
+                    null,
+                    ref dynamicAnalysisSpans,
+                    diagsForCurrentMethod,
+                    ref lazyVariableSlotAllocator,
+                    lambdaDebugInfoBuilder,
+                    closureDebugInfoBuilder,
+                    out stateMachineTypeOpt);
```

Comment:
My understanding is that `LowerBodyOrInitializer` is the main entrypoint to lowering a method body, so even though all of its functionality isn't used.